packages/integration-sdk-cli/src/neo4j/neo4jGraphStore.ts

```diff
@@ -82,21 +85,26 @@ export class Neo4jGraphStore {
         MERGE (${nodeAlias} {_key: $finalKeyValue, _integrationInstanceID: $integrationInstanceID}) 
         SET ${nodeAlias} += $propertyParameters
         SET ${nodeAlias}:${sanitizedType}${classLabels};`;
-      await this.runCypherCommand(buildCommand, {
-        propertyParameters: propertyParameters,
-        finalKeyValue: finalKeyValue,
-        integrationInstanceID: this.integrationInstanceID,
-      });
+      promiseArray.push(
+        this.runCypherCommand(buildCommand, {
+          propertyParameters: propertyParameters,
+          finalKeyValue: finalKeyValue,
+          integrationInstanceID: this.integrationInstanceID,
+        }),
+      );
     }
+    await Promise.all(promiseArray);
```

Comment:
Are we okay with a single failed promise rejecting the main promise?

Reply:
I'd wondered that too.  Technically we should only get a failed promise if the Neo4j bolt driver fails to respond (which should only happen if we lose connection to the Neo4j database, in which case we're in trouble anyway), but it does give me a bit of concern to be honest.